src/Common/tests/System/IO/FileCleanupTestBase.cs

```diff
@@ -78,6 +78,6 @@ protected virtual void Dispose(bool disposing)
         /// <param name="memberName">The member name of the function calling this method.</param>
         /// <param name="lineNumber">The line number of the function calling this method.</param>
         protected string GetTestFileName(int? index = null, [CallerMemberName] string memberName = null, [CallerLineNumber] int lineNumber = 0) =>
-            string.Format(index.HasValue ? "{0}_{1}_{2}" : "{0}_{1}", memberName ?? "TestBase", lineNumber, index.GetValueOrDefault());
+            string.Format(index.HasValue ? "{0}_{1}_{2}_{3}" : "{0}_{1}_{3}", memberName ?? "TestBase", lineNumber, index.GetValueOrDefault(), Guid.NewGuid().ToString("N"));
```

Comment:
> change the tests to use characters that are outside of what GetRandomFileName gives. It creates a base 32 string. (a-z, 0-5) Maybe $?

Other tests fail in such situations, e.g. tests validating special characters in file names.